cmake/onnxruntime_hosting.cmake

```diff
@@ -18,9 +20,11 @@ onnxruntime_add_include_to_target(${PROJECT_NAME} onnxruntime_session gsl)
 
 target_include_directories(${PROJECT_NAME} PRIVATE
         ${ONNXRUNTIME_ROOT}
-        ${Boost_INCLUDE_DIR}
         PUBLIC
-        ${ONNXRUNTIME_ROOT}/hosting/include)
+        ${Boost_INCLUDE_DIR}
```

Comment:
Why make it public?

Reply:
I was trying to propagate the included directories to any targets that use it (e.g. our unit tests) but I couldn't get it working. So for now we need to manually add `${Boost_INCLUDE_DIR}` and the re2 directory to our tests. Ideally, any targets that use this target should be able to find these headers.